zegami_sdk/collection.py

```diff
@@ -223,6 +235,48 @@ def get_image_urls(self, rows, source=0):
             c.HOME, c.API_0, self.workspace_id, self._get_imageset_id(source),
             i) for i in imageset_indices]
 
+    def replace_data(self, data, sheet_name=0):
+        """Replaces the data in the collection.
+
+        The provided input should be a pandas dataframe or a local csv/tsv/xlsx file.
+        Data will be fetched from the default sheet of a xlsx file if sheet_name is not provided.
+        """
+        tsv = ''
+        name = ''
+        if type(data) == pd.DataFrame:
+            tsv = data.to_csv(sep='\t', index=False)
+            name = 'provided_as_dataframe.tsv'
+        else:
+            name = os.path.split(data)[-1]
+            if (file_ext := name.split('.')[-1]) in ['tsv', 'csv']:
+                with open(data, 'r') as f:
+                    tsv = f.read()
+            elif file_ext in ['xls', 'xlsx']:
+                # Not setting the extension to tsv results in error in processing
+                name = name + '.tsv'
+                xlsx_to_pd = pd.read_excel(data, sheet_name=sheet_name)
+                tsv = xlsx_to_pd.to_csv(sep='\t', index=False)
```

Comment:
Not necessary to do this conversion here - in fact better not to! The xls/xlsx file can be uploaded as a blob and cloud deals with reading it, the blob then reflects exactly what the user uploaded.